backends/ubpf/ubpfDeparser.cpp

```diff
@@ -0,0 +1,357 @@
+/*
+Copyright 2019 Orange
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "ubpfDeparser.h"
+#include "ubpfType.h"
+#include "frontends/p4/methodInstance.h"
+
+namespace UBPF {
+
+    class OutHeaderSize final : public EBPF::CodeGenInspector {
+        P4::ReferenceMap*  refMap;
+        P4::TypeMap*       typeMap;
+        const UBPFProgram*  program;
+
+        std::map<const IR::Parameter*, const IR::Parameter*> substitution;
+
+        bool illegal(const IR::Statement* statement)
+        { ::error("%1%: not supported in deparser", statement); return false; }
```

Comment:
you should consider eventually updating to use an ErrType argument.

Reply:
Done.